share/eds/store.go

```diff
@@ -189,7 +193,12 @@ func (s *Store) Remove(ctx context.Context, root share.Root) error {
 //
 // It reads only one quadrant(1/4) of the EDS and verifies the integrity of the stored data by
 // recomputing it.
-func (s *Store) Get(ctx context.Context, root share.Root) (*rsmt2d.ExtendedDataSquare, error) {
+func (s *Store) Get(ctx context.Context, root DataHash) (*rsmt2d.ExtendedDataSquare, error) {
+	// sanity check the root
+	if err := root.Validate(); err != nil {
+		return nil, err
+	}
```

Comment:
Why do some of the methods validate it and some do not?

Another team-wide point. We usually don't do sanity checks for params within internal components and only do those from the user or network inputs. So the Q: should we do it here, or should we trust our networking components that it will do the check?

Reply:
we should have sanity checks at network entrypoints.

Reply:
@renaynay